### PR TITLE
Windows: Add filtering by offset to psscan

### PR DESCRIPTION
**Pull request by adiego8**
Add capability to windows.psscan to filter by offset. The filter would be used by other plugins to find specific offset using the psscan capabilities, like find hidden processes as a result of dkom for example. The `--offset`  flag argument should represent a physical address space, as supported in volatility2.

The `--offset` flag is included in the following plugins:
  - dlllist
  - handles

### CODE REVIEW

**Comment by ikelos** [comment]
Thanks very much for this submission, but it looks like there's a few things about the approach that will need changing.  This currently splits the memory layer to look-up an offset for, from the object that the offset came from.  Each object carries around the layer to which its offset relates to.  For example, the same object (the same data) could be at offset `0x1234` in a virtual layer, but `0x5678` in the physical layer, same object, same data, different layers, and different offsets.  If you then look up offset `0x5678` in the virtual layer, you may find the address doesn't exist.

The way to resolve this is to ask the object you're checking for which layer it's in (you'll need a context to do that, the object you're checking should have one, but otherwise you can ask for that at the filer function call stage), determine if that's a physical layer or a virtual one (this is the hardest part, at the moment we just `isinstance` on `IntelLayer` but given we're trying to add AArch64 support, we'll eventually have to change that, for now stick with it as the best way to determine the physical/virtual split), and then return the result (mapping it, if the layer is in fact virtual).  So instead of a simple lambda, this'll need to return a full function which takes an object, finds the layer, determines if the layer is virtual or not and returns the right offset.  No extra variables (like `memory` required.

Also, `create_offset_filter` should either be parameterized to determine whether it should be filtering on physical or virtual addresses, or be renamed `create_physical_offset_filter`, otherwise it might get confusing in the future.

**Comment by adiego8** [comment]
Appreciate the detailed feedback! I added a new [commit](https://github.com/volatilityfoundation/volatility3/pull/1100/commits/5e38ffa811bcf385c2b470351ed2518118c7d875) that I hope addresses all the points you mentioned above based on my understanding, in case I missed something, do not hesitate to let me know.

**Review by ikelos — APPROVED** [review]
Yep, looks much better, thanks.  5:)

File: volatility3/framework/plugins/windows/psscan.py [hunk 1]
```diff
@@ -59,6 +59,75 @@
...
+                else:
+                    filter_func = (
+                        lambda proc: cls.physical_offset_from_virtual(
+                            context, layer_name, proc
+                        )
+                        != offset
+                    )
+            else:
+                if exclude:
+                    lambda proc: proc.vol.offset == offset
```
[the remark this answers is not in the document]
Comment (ikelos) [reply volatility3/framework/plugins/windows/psscan.py]: So here, you've just made a lambda statement.  I guess you should be assigning it to a variable, so `filter_func =`?

File: volatility3/framework/plugins/windows/psscan.py [hunk 2]
```diff
@@ -59,6 +59,75 @@
...
+                else:
+                    lambda proc: proc.vol.offset != offset
```
[the remark this answers is not in the document]
Comment (ikelos) [reply volatility3/framework/plugins/windows/psscan.py]: Same again.

File: volatility3/framework/plugins/windows/psscan.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import datetime
 import logging
-from typing import Iterable, Callable, Optional, Tuple
+from typing import Iterable, Callable, List, Optional, Tuple
```
[the remark this answers is not in the document]
Comment (ikelos) [reply volatility3/framework/plugins/windows/psscan.py]: Looks like this import isn't necessary after all, if you get rid of it that'll resolve this and we can get it all merged.  5:)

File: volatility3/framework/plugins/windows/psscan.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adiego8) [reply volatility3/framework/plugins/windows/psscan.py]: Good catch, thanks! Commit: https://github.com/volatilityfoundation/volatility3/pull/1100/commits/42c6a9ede6bbe3ff52b868144516679753c9fc8c

File: volatility3/framework/plugins/windows/psscan.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adiego8) [reply volatility3/framework/plugins/windows/psscan.py]: Commit: https://github.com/volatilityfoundation/volatility3/pull/1100/commits/42c6a9ede6bbe3ff52b868144516679753c9fc8c

File: volatility3/framework/plugins/windows/psscan.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (adiego8) [reply volatility3/framework/plugins/windows/psscan.py]: Commit: https://github.com/volatilityfoundation/volatility3/pull/1100/commits/42c6a9ede6bbe3ff52b868144516679753c9fc8c

[the remark this answers is not in the document]
Comment (adiego8) [reply volatility3/framework/plugins/windows/psscan.py]: Would you like me to squash additional commits?

[the remark this answers is not in the document]
Comment (eve-mem) [reply volatility3/framework/plugins/windows/psscan.py]: I know in the past I've not squashed the extra commits - I think that @ikelos is okay with that. Mostly just reviewing the final product. Although I look forward to maybe being corrected by the man himself! :grin:

[the remark this answers is not in the document]
Comment (ikelos) [reply volatility3/framework/plugins/windows/psscan.py]: Yep, no need to squash anything.  Tools are good enough to do the squashing on demand if necessary, and I prefer a full history to see if/how things were changed.  5:)

**Review by ikelos — APPROVED** [review]
Looks good, thanks!

**Pull request merged**